lib/org/chromium/webidl/FetchSpecRunner.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'FetchSpecRunner',
+  extends: 'org.chromium.webidl.PipelineRunner',
+
+  requires: [
+    'foam.net.RetryHTTPRequest',
+    'org.chromium.webidl.HTMLFileContents'
+  ],
+
+  documentation: `A runnable box that expects to receive a PipelineMessage
+    containing an array of URLs. The component will spawn a HTTP worker,
+    which fetches the pages at the given URLs.`,
+
+  constants: {
+    MISSING_URLS: 'URL property of PipelineMessage is missing!',
+  },
+
+  properties: [
+    {
+      documentation: `Name of parser that is used to parse IDL
+        fragments from the fetched HTML files.`,
+      name: 'parser',
+      required: true,
+    },
+    {
+      documentation: `Name of group which the IDL files are canonicalized with.`,
+      name: 'renderer',
```

Comment:
I'd lean toward having an opaque context field in the input that's passed through to the output instead of a single value for all.